Fixed touch area of bottom navigation bar to be filled.

hyochan opened the pull request:
Previously, the top of the touch area of BottomNavigationBarItem is slightly cut off.

Previous result:
![previous](https://user-images.githubusercontent.com/27461460/46496042-f437bd80-c851-11e8-9366-b9f05c1e056b.gif)

Fixed result:
![fixed](https://user-images.githubusercontent.com/27461460/46496054-fd288f00-c851-11e8-93b1-1f07b561ce25.gif)

hyochan committed:
Fixed touch area of bottom navigation bar to be filled.

Previously, the top of touch area of BottomNavigationBarItem is slightly cutted off.

dooboolab commented:
I've submitted `cla`.

tvolkert reviewed (commented):
Thanks for the contribution @dooboolab!

Could you add a test in [bottom_navigation_bar_test.dart](https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/bottom_navigation_bar_test.dart) that would have caught this problem?  It'll ensure we don't accidentally regress it in the future.

tvolkert commented:
/cc @HansMuller

hyochan committed:
Fixed minHeight value for bottomNavigationBar.

Since, giving additionalBottomPadding with symmetric, we need to multiply the value by 2.

hyochan committed:
Renamed additionalBottomPadding to additionalVerticalPadding.

dooboolab commented:
@tvolkert I've like to write up test for this too. However, how can I test if my test code is working correctly? I could fix this problem in `Android Studio` manually editing flutter's code. However, I currently have no idea how to test the flutter itself.

Could you provide me some guideline?

HansMuller commented:
Thanks for your contribution. The before and after animated GIFs are superb!

Unfortunately, we need to address two problems before landing this fix:
- It would be best if we did not change the height of the BottomNavigationBar. Doing so could disturb existing application layouts and tests.
- The custom circle ink splash (`_Circle`) in bottom_navigator.dart no longer matches the [Material spec](https://material.io/design/components/bottom-navigation.html#behavior).  The tap feedback is now supposed to be an [ink ripple](https://docs.flutter.io/flutter/material/InkRipple-class.html).

This is no longer a simple change however if you'd like to attempt it, I can help.

hyochan commented:
I'll try to manage above when my #22956 is merged because current `PR` affect these code.

hyochan commented:
I was going to continue working on this but it seems that this has been taken in #28159.

ryan-sf commented:
> I was going to continue working on this but it seems that this has been taken in #28159.

The ripple effect still does not match the Material spec when I use it.  Are there any examples of this working in the wild?
